runtime/include/chpl-mem-array.h

```diff
@@ -157,6 +168,12 @@ static inline
 void chpl_mem_array_free(void* p,
                          size_t nmemb, size_t eltSize,
                          int32_t lineno, int32_t filename) {
+#ifdef HAS_GPU_LOCALE
+  if (chpl_gpu_is_device_ptr(p)) {
```

Comment:
Just curious -- but do you know how expensive this call is? I imagine it's quite slow.

Reply:
Since in a sublocale configuration we can use wide pointers to indicate which sublocale the pointer is referring to, I don't think we need to use this at all. I would imagine that we already have sublocale `int`s in the wide pointers when this PR and we could check the sublocale field of the wide pointer instead.

Reply:
@ronawho -- My guess was that it would be fast as I was assuming a large chunk of the address space would be used for gpu-accessible memory, and that chunk would be easy to compute. But maybe it is more page-by-page or something, in which case it might be slow.

@mppf -- I think we can do that, but that check will probably have to happen at the module code. Which also is aligned with the general feedback that this whole logic should move to module code.

Reply:
FWIW we do have runtime calls into the generated module code, so you could call it from here even if this part wasn't yet migrated.